docs/teams/intro.md

```diff
@@ -0,0 +1,6 @@
+# Intro to Livebook Teams
+
+[Livebook Teams](https://livebook.dev/teams/?ref=docs) is a product that amplifies Livebook with features designed for teams and businesses.
+
+It's currently in free beta. You can [join the beta waiting list here](https://livebook.dev/teams/?ref=docs) to get early access.
```

Comment:
Anything else we could add to this page to give it more content? Maybe we can duplicate the listing found in the sidebar? 🤔

Reply:
Added the list of features. We may want to add more in the future.